File: src/Entity/Evenements.php
<?php

namespace App\Entity;

use App\Repository\EvenementsRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use ApiPlatform\Core\Annotation\ApiResource;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * @ORM\Entity(repositoryClass=EvenementsRepository::class)
 * @ORM\DiscriminatorMap({"event" = "Evenements"})
 */
class Evenements
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups({"events"})
     */
    private $id;


    /**
     * @ORM\Column(type="date")
     * @Groups({"events"})
     */
    private $Date;

    /**
     * @ORM\Column(type="string", length=50)
     * @Groups({"events"})
     */
    private $adresse;

    /**
     * @ORM\Column(type="string", length=80)
     * @Groups({"events"})
     */
    private $titre;

    /**
     * @ORM\Column(type="string", length=20)
     * @Groups({"events"})
     */
    private $nom;

    /**
     * @ORM\Column(type="integer", nullable=true)
     * @Groups({"events"})
     */
    private $nbrParticipants;

    /**
     * @ORM\Column(type="integer", nullable=true)
     * @Groups({"events"})
     */
    private $nbrMax;

    /**
     * @ORM\Column(type="string", length=255)
     * @Groups({"events"})
     */
    private $img;

    /**
     * @ORM\ManyToOne(targetEntity=Employer::class, inversedBy="evenements")
     * @Groups({"events"})
     */
    private $employer;





    public function getId(): ?int
    {
        return $this->id;
    }



    public function getDate(): ?\DateTimeInterface
    {
        return $this->Date;
    }

    public function setDate(\DateTimeInterface $Date): self
    {
        $this->Date = $Date;

        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(string $adresse): self
    {
        $this->adresse = $adresse;

        return $this;
    }

    public function getTitre(): ?string
    {
        return $this->titre;
    }

    public function setTitre(string $titre): self
    {
        $this->titre = $titre;

        return $this;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getNbrParticipants(): ?int
    {
        return $this->nbrParticipants;
    }

    public function setNbrParticipants(?int $nbrParticipants): self
    {
        $this->nbrParticipants = $nbrParticipants;

        return $this;
    }

    public function getNbrMax(): ?int
    {
        return $this->nbrMax;
    }

    public function setNbrMax(?int $nbrMax): self
    {
        $this->nbrMax = $nbrMax;

        return $this;
    }

    public function getImg(): ?string
    {
        return $this->img;
    }

    public function setImg(string $img): self
    {
        $this->img = $img;

        return $this;
    }

    public function getEmployer(): ?Employer
    {
        return $this->employer;
    }

    public function setEmployer(?Employer $employer): self
    {
        $this->employer = $employer;

        return $this;
    }


}

File: src/Edu/esprit/entities/Evenements.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Edu.esprit.entities;
import java.sql.Date;
import javafx.scene.image.ImageView;

/**
 *
 * @author bensa
 */
public class Evenements {
    
    private int id,nbr_participants,nbr_max,id_emp;
    private String nom,adresse,titre;
    private Date date;
    private String img;
    private ImageView image1;

  
     public ImageView getImage1() {
        return image1;
    }

    public void setImage1(ImageView image1) {
        this.image1 = image1;
    }

    public Evenements() {
        this.nbr_participants=0;
    }

    public Evenements(int id,Date date,String adresse,String titre, String nom,int nbr_participants,int nbr_max,ImageView image1) {
        this.id = id;
        this.nbr_max=nbr_max;
        this.nbr_participants=nbr_participants;
        this.nom = nom;
        this.titre = titre;
        this.adresse=adresse;
        this.date=date;
        this.image1 = image1;

        
    }
    
    public Evenements(int id,Date date,String adresse,String titre, String nom,int nbr_participants,int nbr_max,ImageView image1,int id_emp) {
        this.id = id;
        this.nbr_max=nbr_max;
        this.nbr_participants=nbr_participants;
        this.nom = nom;
        this.titre = titre;
        this.adresse=adresse;
        this.date=date;
        this.image1 = image1;

        
    }
     public Evenements(int id,Date date,String adresse,String titre, String nom,int nbr_participants,int nbr_max,String img) {
        this.id = id;
        this.nbr_max=nbr_max;
        this.nbr_participants=nbr_participants;
        this.nom = nom;
        this.titre = titre;
        this.adresse=adresse;
        this.date=date;
        this.img = img;

        
    }
     
      public Evenements(int id,Date date,String adresse,String titre, String nom,int nbr_participants,int nbr_max,String img,int id_emp) {
        this.id = id;
        this.nbr_max=nbr_max;
        this.nbr_participants=nbr_participants;
        this.nom = nom;
        this.titre = titre;
        this.adresse=adresse;
        this.date=date;
        this.img = img;

        
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getTitre() {
        return titre;
    }
    
    
    public void setTitre(String titre) {
        this.titre = titre;
    }


    public void setAdresse (String adresse) {
        this.adresse = adresse;
    }
    
    public String getAdresse() {
        return adresse;
    }
    
    
    public void setDate (Date date) {
        this.date = date;
    }
    
    public Date getDate() {
        return date;
    }
    
    public void setNbr_participants (int nbr_participants) {
        this.nbr_participants = nbr_participants;
    }
    
    public int getNbr_participants () {
       return nbr_participants;
    }
   
     
    public void setNbr_max (int nbr_max) {
        this.nbr_max = nbr_max;
    }
   
    
    public int getNbr_max () {
       return nbr_max;
    }
    
    
   
public void setImg (String img) {
        this.img = img;
    }
   
    
    public String getImg () {
       return this.img;
    }
    
    
    
   

    
}
